src/core/Elsa.Abstractions/Services/Models/WorkflowExecutionContext.cs

```diff
@@ -32,6 +32,11 @@ public class WorkflowExecutionContext
             IsFirstPass = true;
             Serializer = serviceProvider.GetRequiredService<JsonSerializer>();
             Mediator = serviceProvider.GetRequiredService<IMediator>();
+
+            if (workflowBlueprint.IsDisabled)
+            {
+                WorkflowInstance.WorkflowStatus = WorkflowStatus.Disabled;
+            }
```

Comment:
I will change it to WorkflowInstance.WorkflowStatus = WorkflowStatus.Suspended;
And I will remove WorkflowStatus.Disabled.

Reply:
I was actually thinking you could simply remove these lines. Unless I am missing something here?

Reply:
In the scenario when you start a workflow instance and then while waiting for a signal we disable that workflow and send the signal immediately afterwards we cannot resume the workflow instance. So we have to set some status otherwise I have seen a strange behaviour when I enable the workflow again and run it, but in the result I see that the first instance finished successfully but the last one faulted which does not make any sense.

Reply:
Also actually we need to update the workflow instance status somehow to indicate that it did not finish because the workflow has been disabled before the signal was received.

Reply:
The code above played exactly that role to show user the workflow was disabled (though I know workflow instance may have different status for that scenario). That is why I suggested to use Suspended status.

Reply:
I get it. But I don't like either solution, because a workflow instance at the very least can already infer that its disabled since the workflow blueprint, which it references, is disabled.

To me it makes more sense that the workflow instance should simply not be resumed in the first place. After all, the workflow was disabled, so any workflow instances should be frozen (which also means their state is frozen, so no changes are to be made to its Status field).

Taking your example of the workflow blocked on a signal activity, the code that is responsible for resuming the workflow (which is `WorkflowLaunchpad`) should filter out any matching workflows to execute by checking the disabled state on the associated workflow blueprint.

Once the user enables the workflow blueprint again, then workflow instances can be resumed too if the signal comes in.